Падение старта контекста при встраивании в изолирующий класслоадер (micrometer ThreadLocalAccessor / ServiceLoader)

## Проблема

При встраивании BSL LS через `BSLLSBinding` в хост с изолирующим class loader'ом (thread context classloader не видит классы библиотеки — например, `IsolatedClassloader` сканера SonarQube) Spring-контекст **не поднимается на старте**.

Цепочка падения (создание eager-бина во время `SpringApplication.run()`):

```
UnsatisfiedDependencyException: bean 'sentryAspect'
  → 'sentryExecutor' → 'compositeTaskDecorator' → 'contextPropagatingDecorator'
  → new ContextPropagatingTaskDecorator() → ContextSnapshotFactory.builder()
  → io.micrometer.context.ContextRegistry.<clinit>()  →  loadThreadLocalAccessors()
  → ServiceLoader.load(ThreadLocalAccessor.class)   // однопараметрический → по TCCL
Caused by: java.lang.NoClassDefFoundError: io/micrometer/context/ThreadLocalAccessor
Caused by: java.lang.ClassNotFoundException: io.micrometer.context.ThreadLocalAccessor
  at ...IsolatedClassloader.loadClass
```

`ContextRegistry` при статической инициализации ищет реализации `ThreadLocalAccessor` через `ServiceLoader.load(Class)` (по thread context classloader). Под изолирующим TCCL библиотека micrometer/аксессоры не видны → `<clinit>` падает → падает создание бина декоратора → падает весь старт контекста.

В standalone-режиме (плоский classpath, TCCL видит всё) проблемы нет — она проявляется только при встраивании.

## Как воспроизводится

Обнаружено при поддержке BSL LS в 1c-syntax/sonar-bsl-plugin-community#446: под сканером SonarQube (реальный `IsolatedClassloader`) анализ падает на старте контекста с указанной ошибкой. Сейчас встраивающая сторона вынуждена держать собственный воркэраунд (подмену TCCL вокруг вызова сенсора).

## Предлагаемое решение

Пинить thread context classloader на класслоадер самой библиотеки на время подъёма контекста в `BSLLSBinding.createContext()`. `ContextRegistry` инициализируется один раз и кэшируется, поэтому дальнейшая работа на потоках встраивающего хоста не затрагивается; standalone-поведение не меняется. Это позволит встраивающим убрать собственные обходы.

--- a/src/main/java/com/github/_1c_syntax/bsl/languageserver/binding/BSLLSBinding.java
+++ b/src/main/java/com/github/_1c_syntax/bsl/languageserver/binding/BSLLSBinding.java
@@ -127,6 +127,18 @@ public class BSLLSBinding {
   }
 
   private static ConfigurableApplicationContext createContext() {
-    return getApplication().run();
+    // An embedder may run with a thread context classloader that does not expose this
+    // library's classes (an isolating classloader). micrometer's ContextRegistry discovers
+    // its ThreadLocalAccessor SPI via ServiceLoader on the context classloader during
+    // startup (while the context-propagating task decorator is built), so pin the context
+    // classloader to this library's own classloader for the duration of the boot.
+    var currentThread = Thread.currentThread();
+    var originalClassLoader = currentThread.getContextClassLoader();
+    currentThread.setContextClassLoader(BSLLSBinding.class.getClassLoader());
+    try {
+      return getApplication().run();
+    } finally {
+      currentThread.setContextClassLoader(originalClassLoader);
+    }
   }
 }

